numba/tests/test_unicode.py

```diff
@@ -337,6 +349,41 @@ def test_find(self, flags=no_pyobj_flags):
                                  cfunc(a, substr),
                                  "'%s'.find('%s')?" % (a, substr))
 
+    def test_count(self):
+        pyfunc = count_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES + ['aaaa']:
```

Comment:
I'm not sure that `UNICODE_EXAMPLES` contains suitable material for this test? There's no zero length string, no repeats, no nested repeats. Perhaps take a look at the cpython tests for some inspiration around what would be good to check: https://github.com/python/cpython/blob/5623ac87bbe5de481957eca5eeae06347612fbeb/Lib/test/test_unicode.py#L177-L201

Reply:
That's a good point! I will add some more tests form cpython add improve it.